[docs] Add docker-inside script for wsl2 [skip ci] (#4366)
drud/ddev: docs/content/users/install/ddev-installation.md

    * Installing Chocolatey package manager (optional).
    * One time initialization of mkcert.
    * Installing WSL2 and installing a distro like Ubuntu.

>     * (Optionally) Installing Docker Desktop for Windows and enabling WSL2 integration with the distro.

Reviewer: Move “optional” to the end in the style of the first bullet:

```suggestion
    * Installing Docker Desktop for Windows and enabling WSL2 integration with the distro (optional).
```

Author: ```suggestion
    * Installing Docker Desktop for Windows and enabling WSL2 integration with the distro (optional, do this if you're using the Docker Desktop approach).
```

    * Installing DDEV inside your distro.

>     ### WSL2 with Docker-ce Inside Scripted Installation

Reviewer: Refer to [Docker CE](https://docker-docs.netlify.app/install/), shorten heading, and use the phrase “install script” to be more consistent with [the rest of this page](https://ddev.readthedocs.io/en/stable/users/install/ddev-installation/#system-requirements):

```suggestion
    ### WSL2 + Docker CE Inside Install Script
```

On the fence about the word “inside” here, but I assume it’s critical to point out that Docker CE will live inside the WSL2 distro.

>     This scripted installation prepares your default WSL2 Ubuntu distro for use and has no dependency on Docker Desktop.

Reviewer: Omit “for use” because it’s safe to assume we’re preparing something in order to take further steps with it.

```suggestion
    This scripted installation prepares your default WSL2 Ubuntu distro and has no dependency on Docker Desktop.
```

>     You can do these things manually, or you can do most of it with the provided PowerShell script. 

Reviewer: Start with the preferred/easiest suggestion:

```suggestion
    The provided PowerShell script can do most of the work for you, or you can handle these things manually. 
```

    In all cases:
    

>     1. Install WSL2 with an Ubuntu distro. On a system without WSL2, just run:

Reviewer: [Omit “just”](https://ddev.readthedocs.io/en/stable/developers/writing-style-guide/#avoid-just-and-easy):

```suggestion
    1. Install WSL2 with an Ubuntu distro. On a system without WSL2, run:
```

        ```powershell
        wsl --install
        ```

>         Verify that you have an Ubuntu distro set to default with `wsl -l -v`.

Reviewer: Explicitly clarify that the reader’s supposed to run a command:

```suggestion
        Verify that you have an Ubuntu distro set to default by running `wsl -l -v`.
```

>         If you already have WSL2 but don't have an Ubuntu distro, install one with `wsl --install Ubuntu`. 

Reviewer: Explicitly clarify that the reader’s supposed to run a command:

```suggestion
        If you already have WSL2 but don't have an Ubuntu distro, install one by running `wsl --install Ubuntu`. 
```

        If that doesn't work for you, see the [manual installation](https://docs.microsoft.com/en-us/windows/wsl/install-manual) and linked [troubleshooting](https://docs.microsoft.com/en-us/windows/wsl/troubleshooting#installation-issues).
        

>         If you prefer to use another Ubuntu distro, just install it and set it as default. For example, `wsl --set-default Ubuntu-22.04`.

Reviewer: [Omit “just”](https://ddev.readthedocs.io/en/stable/developers/writing-style-guide/#avoid-just-and-easy):

```suggestion
        If you prefer to use another Ubuntu distro, install it and set it as default. For example, `wsl --set-default Ubuntu-22.04`.
```

    2. Run [this PowerShell script](https://raw.githubusercontent.com/drud/ddev/master/scripts/install_ddev_wsl2_docker_inside.ps1) in an administrative PowerShell Window to complete DDEV WSL2 with Docker Desktop installation:

        ```powershell
        Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; 
        iex ((New-Object System.Net.WebClient).DownloadString('https://raw.githubusercontent.com/drud/ddev/master/scripts/install_ddev_wsl2_docker_inside.ps1'))
        ```

    Now you can use the "Ubuntu" terminal app or Windows Terminal to access your Ubuntu distro, which has DDEV and Docker working in it.

>     ### WSL2 with Docker Desktop Scripted Installation

Reviewer: Match heading style and “install script” with the previous one:

```suggestion
    ### WSL2 + Docker Desktop Install Script
```

    ## Traditional Windows

>     DDEV does work fine on the Windows side, although it’s quite a bit slower than WSL2 by default. You'll want to enable Mutagen or NFS for tolerable performance. Most users find that Mutagen is the most satisfactory technique. See [Performance](performance.md) for more information, but `ddev config global --mutagen-enabled` is the command to enable mutagen.

Reviewer: Get to the point with better links:

```suggestion
    DDEV works fine on the Windows side, but it’s slower than WSL2 by default. Enable either [Mutagen](https://ddev.readthedocs.io/en/stable/users/install/performance/#system-requirements) or [NFS](https://ddev.readthedocs.io/en/stable/users/install/performance/#nfs) for the best performance.
```

In this case, the explanation gets in the way of the instruction:

1. The performance page (and lots of other blurbs) heavily emphasize Mutagen over NFS, so there’s no need to offer commentary here on what most users find best. (We’re leading with that recommendation anyway.)
2. The Mutagen link gets right to the point with that command to enable, so there’s no need to be redundant with it here. (It’s also strangely prescriptive to recommend Mutagen or NFS but only offer the command for Mutagen—might as well link thoughtfully and encourage the reader to either pick the quick + obvious answer or read further and make the choice they’re happiest with. No need to broadly link to the performance page and short-circuit their trip in one breath while distracting from further Windows install notes.)

Author: Links should always be relative in the docs, not link out to readthedocs.

```suggestion
    DDEV works fine on the Windows side, but it’s slower than WSL2 by default. Enable either [Mutagen](performance/#system-requirements) or [NFS](performance/#nfs) for the best performance.```

Reviewer: Yikes, I don’t know what came over me. Thank you for fixing!
